managed/services/inventory/inventory_metrics.go

```diff
@@ -0,0 +1,231 @@
+			prom.BuildFQName(prometheusNamespace, prometheusSubsystem, "services"),
+			"Inventory Service",
+			[]string{"service_id", "service_type", "node_id"},
+			nil),
+
+		metrics: metrics,
+	}
+}
+
+func (i *InventoryMetrics) GetAgentMetrics(ctx context.Context) (metrics []Metric, err error) {
+	metrics = []Metric{}
+
+	err = i.db.InTransactionContext(ctx, nil, func(tx *reform.TX) error {
+		dbAgents, dbAgentsError := models.FindAgents(tx.Querier, models.AgentFilters{})
+
+		if dbAgentsError != nil {
+			return dbAgentsError
+		}
+
+		for _, agent := range dbAgents {
+			disabled := "0"
+			connected := float64(0)
+
+			pmmAgentID := pointer.GetString(agent.PMMAgentID)
+
+			if agent.Disabled {
+				disabled = "1"
+			}
+
+			if i.registry.IsConnected(pmmAgentID) {
+				connected = 1
+			} else {
+				connected = 0
+			}
+
+			agentMetricLabels := []string{
+				agent.AgentID,
+				string(agent.AgentType),
+				pointer.GetString(agent.ServiceID),
+				pointer.GetString(agent.NodeID),
+				pmmAgentID,
+				disabled,
+				pointer.GetString(agent.Version),
+			}
+
+			metrics = append(metrics, Metric{labels: agentMetricLabels, value: connected})
+		}
+		return nil
+	})
+
+	return metrics, err
+}
+
+func (i *InventoryMetrics) GetNodeMetrics(ctx context.Context) (metrics []Metric, err error) {
+	metrics = []Metric{}
+
+	err = i.db.InTransactionContext(ctx, nil, func(tx *reform.TX) error {
+		dbNodes, dbNodesError := models.FindNodes(tx.Querier, models.NodeFilters{})
+
+		if dbNodesError != nil {
+			return dbNodesError
+		}
+
+		for _, node := range dbNodes {
+			nodeMetricLabels := []string{
+				node.NodeID,
+				string(node.NodeType),
+				node.NodeName,
+				pointer.GetString(node.ContainerName),
+			}
+
+			metrics = append(metrics, Metric{labels: nodeMetricLabels, value: serviceEnabled})
+		}
+
+		return nil
+	})
+
+	return metrics, err
+}
+
+func (i *InventoryMetrics) GetServiceMetrics(ctx context.Context) (metrics []Metric, err error) {
+	metrics = []Metric{}
+
+	err = i.db.InTransactionContext(ctx, nil, func(tx *reform.TX) error {
+		dbServices, dbServicesError := models.FindServices(tx.Querier, models.ServiceFilters{})
+
+		if dbServicesError != nil {
+			return dbServicesError
+		}
+
+		for _, service := range dbServices {
+			serviceMetricLabels := []string{
+				service.ServiceID,
+				string(service.ServiceType),
+				service.NodeID,
+			}
+
+			metrics = append(metrics, Metric{labels: serviceMetricLabels, value: serviceEnabled})
+		}
+
+		return nil
+	})
+
+	return metrics, err
+}
+
+func (i *InventoryMetricsCollector) Describe(ch chan<- *prom.Desc) {
+	prom.DescribeByCollect(i, ch)
+}
+
+func (i *InventoryMetricsCollector) Collect(ch chan<- prom.Metric) {
+	ctx, cancelCtx := context.WithTimeout(context.Background(), requestTimeout)
+	defer cancelCtx()
+
+	i.mutex.Lock()
+	defer i.mutex.Unlock()
+
+	agentMetrics, agentError := i.metrics.GetAgentMetrics(ctx)
+
+	if agentError != nil {
+		logrus.Error(agentError)
```

Comment:
please take logger from context

Reply:
what do u mean?

Reply:
see managed/utils/logger

Reply:
but we create own context here
https://github.com/percona/pmm/pull/1854/files#diff-0c01ef95d8b74b9e9f04d3173af69a708935b99a7802facb2209f2cd74c0e04bR191